internal/fnruntime/runner.go

```diff
@@ -423,6 +423,10 @@ func newFnConfig(f *kptfilev1alpha2.Function, pkgPath types.UniquePath) (*yaml.R
 		if err != nil {
 			return nil, errors.E(op, fn, fmt.Errorf("invalid function config %q %w", f.ConfigPath, err))
 		}
+		if node.GetName() == "" {
+			return nil, errors.E(op, fn, fmt.Errorf("functionConfig at path %q is missing `metadata.name`", f.ConfigPath))
```

Comment:
alternative wording inline with http://go/kpt-errors-style-guide:

functionConfig at path %q must have `metadata.name`.

Reply:
Also about code placement, I do remember some code throwing an error related to functionConfig not being a KRM. This check should probably be done at the same place.

Reply:
updated